### PR TITLE
Fix featured post and resource focus styling

### PR DESCRIPTION
**Pull request by ericwbailey**
This PR addresses https://github.com/a11yproject/a11yproject.com/issues/1220. It refactors the featured post and resource components by splitting the concerns up into 3 distinct pieces:

- A featured post partial
- A featured resource partial
- A card partial

While there is a little DRY code between the featured post and resource partials, I think it will be easier to maintain. The previous partial tried to do too much and be too clever about it, which led to causing issues with [the first attempt at a fix](https://github.com/a11yproject/a11yproject.com/pull/1213).

These videos demonstrate the updated fixes:

https://user-images.githubusercontent.com/634191/113942628-1e75e780-97cf-11eb-8a33-04d603f264c9.mov

https://user-images.githubusercontent.com/634191/113942670-2e8dc700-97cf-11eb-9496-afacf5f58a8a.mov

### CODE REVIEW

**Review by SaptakS — COMMENTED** [review]
I have left few minor comments. 

Also, I see that there is now a `_includes/resource.njk` and a `_includes/card/resource.njk`. As far as I can see, the `card/resource.njk` isn't used anywhere at least in this PR. I see that you replaced `post.njk` with `card/post.njk`. Is this intentional?

File: src/_includes/card/resource.njk [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<div class="">
+	<p aria-hidden="true" class="u-font-size-body u-text-transform-uppercase c-banner">
+		<svg aria-hidden="true" focusable="false" class="c-banner__icon" width="22" height="22">
+		{% if resourceType == "Featured" %}
+			<use xlink:href="{{ '/img/icons.svg#featured' | url }}"></use>
```
[the remark this answers is not in the document]
Comment (ericwbailey) [reply src/_includes/card/resource.njk:5]: Yeah, I'm on the same page. I'd like to keep this scoped and address that in a followup PR, if that's okay with you.

File: src/_includes/card/resource.njk [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		{% elif resourceType == "Animation" %}
+			<use xlink:href="{{ '/img/icons.svg#animation' | url }}"></use>
+		{% elif resourceType == "Blogs" %}
+			<use xlink:href="{{ '/img/icons.svg#books' | url }}"></use>
```
[the remark this answers is not in the document]
Comment (ericwbailey) [reply src/_includes/card/resource.njk:9]: Nice catch. Fixing!

**Comment by ericwbailey** [comment]
Thanks for the feedback, @SaptakS. 

>I see that you replaced post.njk with card/post.njk. Is this intentional?

Yes. I've removed `_includes/resource.njk` in favor of `_includes/card/resource.njk`. I think forgetting to remove `_includes/resource.njk` with this PR is part of the reason I was trying to clean up a confusing mess in the first place 😅

Back to you for review.

**Review by SaptakS — APPROVED** [review]
Looks good to me!! Waiting for the CI to pass.

**Pull request merged**